sqlglot/dialects/bigquery.py

```diff
@@ -459,6 +476,13 @@ class BigQuery(Dialect):
 
     EXPRESSION_METADATA = EXPRESSION_METADATA.copy()
 
+    def parse(self, sql: str, **opts) -> t.List[t.Optional[exp.Expression]]:
+        """Override parse to apply SAFE.TIMESTAMP transformation."""
+        expressions = super().parse(sql, **opts)
+
+        # Apply transformation to convert SAFE.TIMESTAMP() to SafeTimestamp
+        return [expr.transform(_transform_safe_timestamp) if expr else expr for expr in expressions]
```

Comment:
We shouldn't do this for two reasons:

1. Implementation-wise, overriding `parse` seems overkill for this one
2. Applying a transformation at every `parse` call to every AST to simply amend a node incurs a significant overhead that is not justified

I think we have an established pattern for dealing with this sort of function, e.g., for `NET.<func>`.